src/unique-code/unique-code.service.ts

```diff
@@ -41,8 +47,13 @@ export class UniqueCodeService {
     await this.uniqueCodeRepository.incrementUsageCount(uniqueCode, em);
 
     if (uniqueCode.email !== email) {
-      throw new BadRequestException("Unique code does not match the provided email");
+      throw new BadRequestException({
+        message: "Unique code does not match the provided email",
```

Comment:
Like above, best to also have `Invalid Unique Code` message here as well. Otherwise, this gives an idea that this code exists but it is not associated with this email.